test/core/promise/example/client.cc

```diff
@@ -0,0 +1,135 @@
+
```

Comment:
@markdroth starting to sketch a full rpc client path

... this certainly doesn't yet compile, but it's of a form that I think it could (and work) from the primitives we have in place